packages/core-transaction-pool/src/memory.ts

```diff
@@ -256,4 +273,16 @@ export class Memory {
 
         return removed;
     }
+
+    private currentHeight(): number {
+        if (app.has("state")) {
+            return app.resolve("state").getLastBlock().data.height;
+        }
+
+        if (app.has("blockchain")) {
```

Comment:
This will never be reached once `core-blockchain` is mounted because `if (app.has("state"))` should always be `true` at that point. You have to chose between `getLastHeight` or `getLastBlock`.

Reply:
Even better! Removed.